routes/items.js

```diff
@@ -1,31 +1,58 @@
 const { Router } = require("express");
 const router = Router();
-
 const itemData = require('../dataInterface/items');
 
+// This NPM Package aid in parsing the request body
+const bodyParser = require('body-parser');
+// router?app.use(something()) tells the server to use this function throug it
+router.use(bodyParser())
+
 router.get("/", (req, res, next) => {
   res.status(200).send(itemData.getAll())
 });
 
 router.get("/:id", (req, res, next) => {
-  // TODO: complete writing this route handler
-  res.status(501).send({ error: 'route not yet implemented' });
+  const queryExist = itemData.getById(req.params.id)
+  if(queryExist){
+    res.status(200).json(queryExist)
+  }else{
+    res.status(404).send({error: "not found"})
+  }
 });
 
 router.post("/", (req, res, next) => {
-  itemData.create(req.body);
-  res.sendStatus(200);
+  if(req.body.field != ""){
```

Comment:
Nice check! In the case that the body data is incorrect, the 402 HTTP status is the most appropriate (https://httpwg.org/specs/rfc9110.html#status.422), though 400 is okay. 

For absolute error proofing, consider also adding a check for the return value from `itemData.create()`. If it was connected to a real database, it would not return a nicely formed new object if things went wrong on creation. It may instead return an error, error message, or `null`.